### PR TITLE
UI artifacts list improvements

### PR DESCRIPTION
**Pull request by lotas**
* fix copy URL functionality for some artifacts (#5348)
* allow to run local UI against remote Taskcluster instance (#5373)
* put log files on top of the list (#5362)

Closes bugs: #5348 #5373 #5362

image

### CODE REVIEW

File: ui/src/components/TaskRunsCard/sort.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  return artifacts
+    .map(a => ({ ...a, priority: getPriority(a) }))
+    .sort((a, b) => {
+      if (a.priority === b.priority) {
+        return a.node?.name?.localeCompare(b.node?.name);
+      }
+
+      return a.priority - b.priority;
+    });
+};
```
Comment (matt-boris) [inline ui/src/components/TaskRunsCard/sort.js:23]: Love this!

File: ui/README.md [hunk 2]
```diff
@@ -19,6 +19,15 @@ http://localhost:3050, but otherwise need not be publicly accessible. The
 development server for this repo will proxy requests as necessary to
 http://localhost:3050.
 
+### Using existing Taskcluster server
```
Comment (matt-boris) [inline ui/README.md:22]: 🎉

**Review by matt-boris — APPROVED** [review]
Awesome! LGTM

File: ui/src/components/TaskRunsCard/sort.js (hunk 1 above)
Comment (lotas) [reply ui/src/components/TaskRunsCard/sort.js:23]: thanks! Spend some time trying to figure out how to do it with `.sort()` only, but didn't find a good solution. Had to attach those virtual priorities

File: ui/src/components/TaskRunsCard/index.jsx [hunk 3]
```diff
@@ -252,13 +253,7 @@ export default class TaskRunsCard extends Component {
 
     return {
       ...artifacts,
-      edges: [...artifacts.edges].sort((a, b) => {
-        if (a.node.isPublic === b.node.isPublic) {
```
Comment (lotas) [inline ui/src/components/TaskRunsCard/index.jsx:256]: btw, search code base didn't revel any place where this value was set. There were only some checks on the queue api side, but it was never exposed, so I wonder if it ever worked at all

Comment (djmitche) [reply ui/src/components/TaskRunsCard/index.jsx:256]: I recall this property (`isPublic`) was calculated in a weird way in a weird place, and I think I removed that.  Maybe I missed this spot.

Comment (lotas) [reply ui/src/components/TaskRunsCard/index.jsx:256]: Thanks! I decided that calculating it in the UI is enough for now, so didn't change API returned object.

**Pull request merged**